.ci/local_integration_test

```diff
@@ -31,6 +64,140 @@ if ! hash ginkgo; then
+    echo "Enter the project"
+    read PROJECT
+    echo "Enter the control cluster"
+    read CONTROL_CLUSTER
+    echo "Enter the target cluster"
+    read TARGET_CLUSTER
+    echo "Enter the machine class path"
+    read MACHINECLASS
+    if [ ! -f "$MACHINECLASS" ]
+    then
+        printf "\e[31m$MACHINECLASS is not a valid file\e[0m\n"
+        exit 1
+    fi
+    make non-gardener-setup
+    cd -
+    cd ../../..
+    printf "\n%s" "MACHINECLASS_V1=${MACHINECLASS}" >> ".env"
+    cd test/integration/controller
+fi
+
+cd ../../..
+set -o allexport
+source .env
+set +o allexport
+cd test/integration/controller
+
+CREDENTIALS_SECRET_NAME=shoot-operator-aws-team
+GARDEN_CORE_NAMESPACE=garden-core
+
+function fetch_credentials(){
+   kubectl  get secret $CREDENTIALS_SECRET_NAME -n $GARDEN_CORE_NAMESPACE -o yaml > credentials.yaml
+   accessKeyID=$(yq eval '.data.accessKeyID' credentials.yaml)
+   secretAccessKey=$(yq eval '.data.secretAccessKey' credentials.yaml)
+   echo $accessKeyID
+   echo $secretAccessKey
+}
+
+function fetch_userdata(){
+    NAMESPACE='garden'
+    USERDATA_SECRET_NAME=$(kubectl get secrets -o custom-columns=Name:.metadata.name --no-headers | grep shoot--$PROJECT | head -n 1)
+    kubectl  get secret $USERDATA_SECRET_NAME  -oyaml > userData.yaml
+    userData=$(yq eval '.data.userData' userData.yaml)
+    echo $userData
+}
+
+function create_test_mc_secret() {
+    gardenctl target --garden sap-landscape-dev
+    eval $(gardenctl kubectl-env bash)
+    result=$(fetch_credentials)
+    accessKeyID=$(echo "$result" | head -n 1)
+    secretAccessKey=$(echo "$result" | tail -n 1)
+    gardenctl target --garden sap-landscape-dev --project $PROJECT --shoot $TARGET_CLUSTER --control-plane
+    eval $(gardenctl kubectl-env bash)
+    userData=$(fetch_userdata)
+    rm -f credentials.yaml
+    rm -f userData.yaml
+secret_yaml=$(cat <<EOF
+apiVersion: v1
+data:
+  accessKeyID: $accessKeyID
+  secretAccessKey: $secretAccessKey
+  userData: $userData
+kind: Secret
+metadata:
+  name: test-mc-secret
+  namespace: default
+type: Opaque
+EOF
+)
+
+    gardenctl target --garden sap-landscape-dev --project $PROJECT --shoot $CONTROL_CLUSTER
+    eval $(gardenctl kubectl-env bash)
+    echo "$secret_yaml" | kubectl --kubeconfig=$CONTROL_KUBECONFIG apply -f -
+}
+
+if [ "$IS_CONTROL_CLUSTER_SEED" = false ]
+then
+  create_test_mc_secret
+  printf "\e[32mSuccessfully created test_mc_secret on control cluster\e[0m\n"
+fi
+
+printf "\e[33mStarting integration tests...\e[0m\n"
+
+set +e
 ginkgo -v --show-node-events --poll-progress-after=300s --poll-progress-interval=60s
+set -e
+printf "\e[32mIntegration tests completed successfully\e[0m\n"
+
+if [ "$IS_CONTROL_CLUSTER_SEED" = false ]
+then
+  set +e
+  kubectl --kubeconfig=$CONTROL_KUBECONFIG delete secret test_mc_secret
+  set -e
+  printf "\e[32mSuccessfully deleted test_mc_secret on control cluster\e[0m\n"
+fi
+
+cd ../../..
+rm -f .env
+cd test/integration/controller
```

Comment:
Is this required, will it make a difference?

Reply:
actually this file is generate by this script only, so better to clean it up.

Reply:
I just meant the last line
is it necessary to change directories at the end?

Reply:
not necessary but just for reseting the working directory to the one from which user invoked the script.